src/code/actions.js

```diff
@@ -271,7 +272,9 @@ function _submitDrake(targetDrakeIndex, userDrakeIndex, correct, state) {
         userDrakeOrg = GeneticsUtils.convertDrakeToOrg(state.drakes[userDrakeIndex]),
         initialDrakeOrg = state.initialDrakes[userDrakeIndex] ? GeneticsUtils.convertDrakeToOrg(state.initialDrakes[userDrakeIndex]) : null,
         routeSpec = state.routeSpec,
-        editableGenes = state.authoring[routeSpec.level][routeSpec.mission][routeSpec.challenge].visibleGenes.split(", ");
+        challengeDefinition = AuthoringUtils.getChallengeDefinition(state.authoring, routeSpec),
+        editableGeneString = challengeDefinition.visibleGenes || challengeDefinition.userChangeableGenes,
```

Comment:
Why are the visible genes the default ones we're submitting for "editableGenes"? If a challenge has `editableGenes: "tail", visibleGenes: "wings"`, then this means that the user can change "tail", and sees "wings" as a static, non-editable gene. With this, we'd just be submitting "wings".